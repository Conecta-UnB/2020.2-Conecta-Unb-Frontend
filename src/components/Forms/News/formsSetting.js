import * as Yup from 'yup';

export const initialValues = {
  nome: '',
  matricula: '',
  email: '',
  senha: '',
  telefone: '',
};

export const SignupSchema = Yup.object().shape({
  titulo: Yup.string().required('É necessário informar um título'),
  autor: Yup.string().required('É necessário indicar um autor'),
  imagem: Yup.string(),
  conteudo: Yup.string().required('É necessário indicar o conteúdo'),
});

export const handleSubmit = async (values, setSubmitting, createNews) => {
  setSubmitting(false);
  await createNews({
    titulo: values.titulo,
    autor: values.autor,
    imagem: values.imagem,
    conteudo: values.conteudo,
  });
};
