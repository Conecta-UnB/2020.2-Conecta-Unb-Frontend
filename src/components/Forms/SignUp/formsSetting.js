import * as Yup from 'yup';

export const initialValues = {
  nome: '',
  matricula: '',
  email: '',
  senha: '',
  telefone: '',
};

export const SignupSchema = Yup.object().shape({
  nome: Yup.string().required('Required'),
  matricula: Yup
    .string()
    .length(9, 'Insira a matricula sem o /, total de 9 números')
    .matches(/^\d+$/, 'A matricula deve ser um número')
    .required('É necessário indicar uma matricula válida'),
  email: Yup
    .string()
    .email('Email inváilido')
    .required('É necessário indicar um email'),
  telefone: Yup
    .string()
    .length(11, 'O telefone deve ter 11 digitos')
    .matches(/^\d+$/, 'O telefone deve ser um número')
    .required('É necessário indicar um número'),
  senha: Yup
    .string()
    .min(6, 'A senha deve ter pelo menos 6 dígitos')
    .required('É necessário criar uma senha'),

});

export const handleSubmit = async (values, setSubmitting, createUser) => {
  setSubmitting(false);
  await createUser({
    nome: values.nome,
    matricula: values.matricula,
    email: values.email,
    senha: values.senha,
    telefone: values.telefone,
  });
};
