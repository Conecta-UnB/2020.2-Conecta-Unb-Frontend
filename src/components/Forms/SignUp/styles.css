.containerSignUpForm {
  width: 90%;
}

.containerSignUpForm form{
  flex-direction: column;
  display: flex;
  align-items: center;
  justify-content: center;
}

.containerSignUpForm input{
  width: 90%;
  height: 50px;
  border-radius: 10px;
  font-size: 24px;
  border-color: #018B44;
  color: #018B44;
  margin-top: 3%;
  padding: 1%;
}

.containerSignUpForm button{
  margin-top: 3%;
  padding: 1%;
  background: #018B44;
  border: none;
  color: #ffff;
  width: 25%;
}

.errorField {
  margin-top: 15px;
  font-size: 16px;
  color: #c50404;
}