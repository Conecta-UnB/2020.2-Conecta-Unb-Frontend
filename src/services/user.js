export async function createUser(Host, user) {
  try {
    const response = await fetch(`${Host}/api/user/create`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        nome: user.nome,
        matricula: user.matricula,
        email: user.email,
        senha: user.senha,
        telefone: user.telefone,
      }),
    });
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log('error ', error);
    return error;
  }
}
export async function getUser(Host, matricula) {
  const response = await fetch(`${Host}/api/user/${matricula}`);
  try {
    const data = await response.json;
    return data.data.user;
  } catch (error) {
    return error;
  }
}
