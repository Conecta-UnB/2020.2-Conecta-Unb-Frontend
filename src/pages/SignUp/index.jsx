import React from 'react';
import SignUpForm from '../../components/Forms/SignUp';
import signUp from '../../assets/signUp.svg';
import './styles.css';

export default function SignUpPage() {
  return (
    <div>
      <div className="underHeader">
        <h1>
          Cadastro
        </h1>
        <div id="formSignUp">
          <SignUpForm />
          <img src={signUp} alt="signUpImage" />
        </div>
      </div>
    </div>
  );
}
