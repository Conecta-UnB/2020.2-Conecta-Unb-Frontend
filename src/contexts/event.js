import React, { createContext, useState } from 'react';
import * as EventService from '../services/event';

export const EventContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function EventProvider({ children }) {
  const [allEvents, setAllEvents] = useState('');
  const [events] = useState('');
  const Host = 'http://localhost:3333';

  async function createEvent(event) {
    console.log('ola', event);
    const response = await EventService.createEvent(Host, event);
    return response;
  }
  async function updateEvent(event) {
    console.log('ola', event);
    const response = await EventService.updateEvent(Host, event);
    return response;
  }

  async function deleteEvent(event) {
    console.log('ola', event);
    const response = await EventService.deleteEvent(Host, event);
    return response;
  }

  async function readEvent(event) {
    console.log('ola', event);
    const response = await EventService.readEvent(Host, event);
    return response;
  }

  async function readAllEvents() {
    const response = await EventService.readAllEvents(Host);
    setAllEvents(response.data);
    console.log(response.data);
    console.log('estou aqui');
    return response;
  }

  return (
    <EventContext.Provider
      value={{
        events,
        allEvents,
        createEvent,
        updateEvent,
        deleteEvent,
        readAllEvents,
        readEvent,
      }}
    >
      {children}
    </EventContext.Provider>
  );
}
