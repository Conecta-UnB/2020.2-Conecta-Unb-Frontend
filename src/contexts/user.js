import React, { createContext } from 'react';
import * as UserService from '../services/user';

export const UserContext = createContext({});

// eslint-disable-next-line react/prop-types
export default function UserProvider({ children }) {
  // const [user, setUser] = useState({});
  const Host = 'http://localhost:3333';
  async function createUser(user) {
    console.log('ola', user);
    const response = await UserService.createUser(Host, user);
    return response;
  }

  async function login(user) {
    const response = await UserService.login(Host, user);
    return response;
  }

  return (
    <UserContext.Provider
      value={{
        createUser,
        login,
      }}
    >
      {children}
    </UserContext.Provider>
  );
}
