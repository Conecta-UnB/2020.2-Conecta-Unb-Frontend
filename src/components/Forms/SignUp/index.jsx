import React, { useState, useContext } from 'react';
import './styles.css';
import { UserContext } from '../../../contexts/user';

export default function SignUpForm() {
  const [nome, setNome] = useState('');
  const [matricula, setMatricula] = useState('');
  const [email, setEmail] = useState('');
  const [senha, setSenha] = useState('');
  const [telefone, setTelefone] = useState('');

  const { createUser } = useContext(UserContext);

  async function handleAdd(e) {
    e.preventDefault();
    console.log(nome, matricula, email);
    await createUser({
      nome, matricula, email, senha, telefone,
    });
  }

  return (
    <div className="containerSignUpForm">
      <form onSubmit={handleAdd}>
        <input type="text" autoComplete="off" placeholder="Nome completo" value={nome} onChange={(e) => { setNome(e.target.value); }} />
        <input type="text" autoComplete="off" placeholder="MatrÃ­cula" value={matricula} onChange={(e) => { setMatricula(e.target.value); }} />
        <input type="email" autoComplete="off" placeholder="E-mail" value={email} onChange={(e) => { setEmail(e.target.value); }} />
        <input type="password" autoComplete="off" placeholder="Senha" value={senha} onChange={(e) => { setSenha(e.target.value); }} />
        <input type="text" autoComplete="off" placeholder="Telefone" value={telefone} onChange={(e) => { setTelefone(e.target.value); }} />
        <button type="submit">
          Cadastre-se
        </button>
      </form>
    </div>
  );
}
