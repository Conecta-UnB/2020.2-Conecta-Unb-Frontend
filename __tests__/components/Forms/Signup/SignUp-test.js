import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SignUpForm from '../../../../src/components/Forms/SignUp';
import * as formSettings from '../../../../src/components/Forms/SignUp/formsSetting';
import { UserContext } from '../../../../src/contexts/user';

describe('testing loginforms', () => {
  test('deve ser igual a snapshot', () => {
    const { asFragment } = render(<SignUpForm />);
    expect(asFragment(<SignUpForm />)).toMatchSnapshot();
  });

  test('test if the form works', async () => {
    const createUser = jest.fn();
    formSettings.handleSubmit = jest.fn();
    render(
      <UserContext.Provider value={{ createUser }}>
        <SignUpForm />
      </UserContext.Provider>,
    );
    userEvent.type(screen.getByPlaceholderText('MatrÃ­cula'), '187362579');
    userEvent.type(screen.getByPlaceholderText('E-mail'), 'senhor_teste@gmail.com');
    userEvent.type(screen.getByPlaceholderText('Senha'), 'amoASenhoraTeste');
    userEvent.type(screen.getByPlaceholderText('Telefone'), '61993211456');
    userEvent.type(screen.getByPlaceholderText('Nome completo'), 'Senhor teste');
    userEvent.click(screen.getByRole('button', { name: /Cadastre-se/i }));
    await waitFor(() => {
      expect(formSettings.handleSubmit).toHaveBeenCalledTimes(1);
      expect(formSettings.handleSubmit).toHaveBeenCalledWith({
        email: 'senhor_teste@gmail.com',
        matricula: '187362579',
        nome: 'Senhor teste',
        senha: 'amoASenhoraTeste',
        telefone: '61993211456',
      },
      expect.any(Function),
      expect.any(Function));
    });
  });
});
