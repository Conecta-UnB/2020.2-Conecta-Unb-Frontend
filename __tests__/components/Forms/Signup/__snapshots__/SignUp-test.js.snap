// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing loginforms deve ser igual a snapshot 1`] = `
<DocumentFragment>
  <div
    class="containerSignUpForm"
  >
    <form
      action="#"
    >
      <input
        name="nome"
        placeholder="Nome completo"
        value=""
      />
      <input
        name="matricula"
        placeholder="MatrÃ­cula"
        value=""
      />
      <input
        name="email"
        placeholder="E-mail"
        type="email"
        value=""
      />
      <input
        name="senha"
        placeholder="Senha"
        type="password"
        value=""
      />
      <input
        name="telefone"
        placeholder="Telefone"
        value=""
      />
      <button
        type="submit"
      >
        Cadastre-se
      </button>
    </form>
  </div>
</DocumentFragment>
`;
